qa/tasks/cephfs/test_cephfs_shell.py

```diff
@@ -816,6 +821,128 @@ def test_issue_cephfs_shell_cmd_at_invocation(self):
                    "expected_output -\n{}\ndu_output -\n{}\n".format(
                    dirname, output)
 
+    def test_every_shell_cmd_at_invocation(self):
+        """
+        Runs every test in this suite with
+        ``cephfs-shell -c ceph.conf <cephfs-shell-command>``
+        instead of ``cephfs-shell -c ceph.conf -- <cephfs-shell-command>``
+        """
+        from sys import stdout as sys_stdout
+        from sys import modules as sys_modules
+        from inspect import isclass as inspect_isclass
+        from inspect import ismethod as inspect_ismethod
+        from inspect import getmembers as inspect_getmembers
+        from traceback import print_exc as traceback_print_exc
+
+        def fake_run_cephfs_shell_cmd(cmd, mount_x=None, opts=None,
+                                      config_path=None, stdin=None):
+            if mount_x is None:
+                mount_x = self.mount_a
+            if config_path is None:
+                config_path = self.mount_a.config_path
+
+            cmd = " ".join(cmd) if isinstance(cmd, list) else cmd
+
+            args = ["cephfs-shell", "-c", self.mount_a.config_path]
+
+            if opts is not None:
+                args.extend(opts)
+
+            args.append(cmd)
+
+            log.info("Running command: {}".format(" ".join(args)))
+            return mount_x.client_remote.run(args=args, stdout=StringIO(),
+                                             stderr=StringIO(), stdin=stdin)
+
+        def fake_get_cephfs_shell_cmd_output(cmd, mount_x=None, opts=None,
+                                        stdin=None, config_path=None):
+            return fake_run_cephfs_shell_cmd(cmd=cmd, mount_x=mount_x,
+                                             opts=opts, stdin=stdin,
+                                             config_path=config_path).\
+                stdout.getvalue().strip()
+
+        def fake_get_cephfs_shell_cmd_error(cmd, mount_x=None, opts=None,
+                                            stdin=None, config_path=None):
+            return fake_run_cephfs_shell_cmd(cmd=cmd, mount_x=mount_x,
+                                             opts=opts, stdin=stdin,
+                                             config_path=config_path).\
+                stderr.getvalue().strip()
+
+        def discover_classes_and_run_tests():
```

Comment:
Please comment what these tests are doing. I'm not really sure.

Reply:
[test_every_shell_cmd_at_invocation](https://github.com/ceph/ceph/pull/32288/files#diff-0176720ea87b333b7c452e72f8adcf42R824) collects all tests from test_cephfs_shell.py and tests them with non-interactive shell this time.

Reply:
> Please comment

add to the code!

Reply:
Present already - https://github.com/ceph/ceph/pull/32288/files#diff-0176720ea87b333b7c452e72f8adcf42R826